### DISCUSSION TITLE
[HydroDyn] How to generate wave elevation at a user-defined location instead of at (0, 0) when WaveMod = 5.

### DISCUSSION THREAD

**Discussion by JT-Kim89** [OpenFAST/openfast#2567] [Q&A]
Hi, 

I am currently comparing results with model tests and would like to generate external waves using `WaveMod = 5` based on the location of the wave gauge. At present, it seems that external waves can only be imported relative to the (0, 0) position. Is there a way to import external waves based on a user-defined location instead? For example, I would like to input wave data measured at locations such as (-10, 0) or (0, 10) instead of the default (0, 0).

Thank you.

**Comment by jjonkman** [comment]
Dear @JT-Kim89,

The `WaveMod` = 5 (user-defined wave-elevation time series) option of HydroDyn (in OpenFAST 3.5.4 and older) and SeaState (in OpenFAST v4.0 and newer) currently requires that you specify the wave-elevation time series at (X,Y) = (0,0) in global coordinates.  Often in model tests, separate wave-only tests (without the model present) are run with the wave probe at the center where the model will be placed (0,0) to enable post-processing.

If that is not available for your model tests, you should be able to use the linear wave-dispersion relationship to estimate the wave elevation time series at (0,0) from the wave elevation time series you know at another location (which is what SeaState/HydroDyn will solve internally to compute the wave elevation at other locations based on what is provided by the user at (0,0)).

Best regards,

**Reply by JT-Kim89** [reply]
Dear @jjonkman,

Thank you for your kind explanation. Based on the linear-wave-dispersion relationship you mentioned, I will proceed to generate the wave elevation at the (0, 0) location.

Sincerely,
